lib/charms/data_platform_libs/v0/database_provides.py

```diff
@@ -0,0 +1,293 @@
+# Copyright 2022 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Relation provider side abstraction for database relation.
+
+This library is mostly an uniform interface to a selection of common databases
+metadata, with added custom events that add convenience to database management,
+and methods to set the application related data.
+
+Following an example of using the DatabaseRequestedEvent, in the context of the
+database charm code:
+
+```python
+from charms.data_platform_libs.v0.database_provides import DatabaseProvides
+
+class SampleCharm(CharmBase):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        # Charm events defined in the database provides charm library.
+        self.provided_database = DatabaseProvides(self, "database")
```

Comment:
I'd either make this explicit, or drop it in favour of the default value.

```suggestion
        self.provided_database = DatabaseProvides(self, relation_name="database")
```

Reply:
Thanks. Updated on https://github.com/canonical/data-platform-libs/pull/2/commits/b51cb789ef94f9741bb2c8190b2fda64028f10db.